### PR TITLE
Update supported Clang (CUDA), Intel, and ROCM compiler versions

### PR DESCRIPTION
**Pull request by ajpowelsnl**
This PR updates the Clang for CUDA , ROCM and Intel compiler versions.

### CODE REVIEW

**Review by crtrott — APPROVED** [review]

File: docs/source/requirements.rst [hunk 1]
```diff
@@ -5,8 +5,7 @@ Compiler Versions
 =================
 
 Generally Kokkos should work with all compiler versions newer than the minimum.
-However as in all sufficiently complex enough code, we have to work around compiler
-bugs with almost all compilers. So compiler versions we don't test may have issues
+However,in complex code, we have to work around compiler bugs. So compiler versions we don't test may have issues
```
Comment (PhilMiller) [inline docs/source/requirements.rst:8]: There's a missing space after "However,"

Comment (dalg24) [reply docs/source/requirements.rst:8]: Too slow :/

File: docs/source/testing-and-issue-tracking/Testing-Process-Details.md [hunk 2]
```diff
@@ -12,14 +12,11 @@ The following list is taken from file kokkos/README in the Kokkos Github reposit
...
     * Intel 17.4.196
     * Intel 18.0.128
-    * Clang 3.6.1
-    * Clang 4.0.0
-    * Clang 4.0.0 for CUDA (CUDA Toolkit 8.0.44)
+    * Clang 4.0.0 for CPU
+    * Clang 10.0.0 for CUDA (CUDA Toolkit 11.0.1)
     * PGI 18.10
     * NVCC 7.0 for CUDA (with gcc 4.8.4)
     * NVCC 7.5 for CUDA (with gcc 4.8.4)
```
Comment (PhilMiller) [inline docs/source/testing-and-issue-tracking/Testing-Process-Details.md:22]: These NVCC lines (here and just below) are out of date too - it should be 9.2, and 10.x and 11.y for some values of x and y

File: docs/source/ProgrammingGuide/Compiling.md [hunk 3]
```diff
@@ -19,11 +19,12 @@ To compile Kokkos, a C++14 compliant compiler is needed. For an up-to-date list
 Minimum Compiler Versions
 
     GCC: 5.3.0
-    Clang: 4.0.0
+    Clang: 4.0.0  (CPU)
+    Clang: 10.0.0 (as CUDA compiler)
```
[the remark this answers is not in the document]
Comment (masterleinad) [reply docs/source/ProgrammingGuide/Compiling.md:23]: What about `IntelLLVM`, `NVHPC`?

**Pull request merged**